fsw/src/crypto_props.c

```diff
@@ -0,0 +1,351 @@
+/*
+ This implementation file implements functions from crypto_props.h
+ * which are required for key,value property management via linked list
+ * data structure.  
+ */
+/* 
+ * File:   crypto_props.c
+ * Author: Ary Naim (aryan.e.naim@jpl.nasa.gov)
+ *
+ * Created on November 4, 2021, 9:37 AM
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include "stdbool.h"
+#include <string.h>
+#include "crypto_props.h"
+
+crypto_config_list* p_self = NULL; 
+/*=============================================================================
+Private Declarations
+==============================================================================*/
+int crypto_config_props_set_property_value_helper(char* key, char* value, crypto_config_list** props);
+char* crypto_config_props_get_property_value_helper(char* key, crypto_config_list** props);
+int crypto_config_props_add_property_helper(char* key, char* value,crypto_config_list** pp_self); 
+int crypto_config_props_remove_property_helper(char* key,crypto_config_list** pp_self); 
+int safe_copy(char* dest, char* src);
+/*=============================================================================
+ * Implementations
+==============================================================================*/
+crypto_config_list* crypto_config_alloc(int max_key_size, int max_value_size)
+{
+    if (NULL==p_self)
+    {
+        p_self = (crypto_config_list*) calloc(1,sizeof(crypto_config_list));
+        p_self->KEY_SIZE_IN_BYTES = max_key_size; 
+        p_self->VALUE_SIZE_IN_BYTES = max_value_size; 
+        
```

Comment:
I'd actually prefer key/value sizes dynamically allocated based on the actual keys/values that get read in instead of hardcoded sizes... Again, this won't hold up merging this branch since we want to start using this now, but that should be a future improvement.

Reply:
Will be added in future versions, I added this to TODO list.